### PR TITLE
[#9841] Clean up type checking/hinting for twisted.web.http_headers

### PR DESCRIPTION
**Pull request by hawkowl**
https://twistedmatrix.com/trac/ticket/9841

### CODE REVIEW

File: src/twisted/web/http_headers.py [hunk 1]
```diff
@@ -208,20 +209,35 @@ def setRawHeaders(self, name, values):
         @param values: A list of strings each one being a header value of
             the given name.
 
+        @raise TypeError: Raised if C{values} is not a L{list} of L{bytes}
+            or L{unicode} strings, or if C{name} is not a L{bytes} or
+            L{unicode} string.
```
Comment (twm) [inline src/twisted/web/http_headers.py:214]: Should we still be documenting things as `unicode` in this brave new Python 3 world? I was tempted to change these in my last PR.

Comment (wsanchez) [reply src/twisted/web/http_headers.py:214]: We should switch to `str`.  There is no ambiguity left there.

**Comment by rodrigc** [comment]
@wsanchez @twm I continued this PR at https://github.com/twisted/twisted/pull/1389

**Comment by rodrigc** [comment]
Fixed in https://github.com/twisted/twisted/pull/1389

**Pull request closed without merging**